gradle/publish.gradle

```diff
@@ -39,13 +39,36 @@ def selectArtifactId(project, type, defaultName) {
     }
 }
 
+def isAvailableForPublication(publication) {
+    def name = publication.name
+    def result = false
+    result = result | (name in ['jvm', 'js', 'metadata', 'kotlinMultiplatform'])
+    result = result | (project.ext.isLinuxHost && name == 'linuxX64')
+    result = result | (project.ext.isWinHost && name == 'mingwX64')
+    result = result | (project.ext.isMacosHost && name in ['iosX64, iosArm32, iosArm64', 'macosX64'])
+
+    return result
+}
+
+tasks.withType(AbstractPublishToMaven).all {
+    onlyIf { isAvailableForPublication(publication) }
+}
+
 publishing {
     repositories {
         maven {
             if (publishLocal) {
                 url globalM2
             } else {
-                url 'https://dl.bintray.com/kotlin/ktor'
+                def user = 'kotlin'
+                def repo = 'ktor'
+                def name = 'ktor'
+                url = "https://api.bintray.com/maven/$user/$repo/$name/;publish=0"
```

Comment:
Btw, we probably need to extract it (or the part user/repo) to `gradle.properties` to be able to easily amend it or override by setting `-P...`